iOSClient/Networking/NCNetworking.swift

```diff
@@ -183,10 +185,14 @@ import Queuer
                 // write certificate tmp to disk
                 certificate.write(toFile: directoryCertificate + "/" + host + ".tmp", atomically: true)
                 
-                // verify
-                let certificateSavedPath = directoryCertificate + "/" + host + ".der"
-                if let certificateSaved = NSData(contentsOfFile: certificateSavedPath), certificate.isEqual(to: certificateSaved as Data) {
+                if isServerTrusted {
                     return true
+                } else if status == errSecSuccess {
+                    // verify
+                    let certificateSavedPath = directoryCertificate + "/" + host + ".der"
+                    if let certificateSaved = NSData(contentsOfFile: certificateSavedPath), certificate.isEqual(to: certificateSaved as Data) {
+                        return true
+                    }
                 }
```

Comment:
```suggestion
                    } // else: certificate not trusted, fallthrough
```
Ending with an `else-if` is a code smell. At the very least add a comment as to why there is no final `else`

Reply:
No finished yet, ...